### PR TITLE
Display CORS Headers for Non-Options Requests (Review)

### PR DESCRIPTION
**Pull request by dubeejw**
Fixes: https://github.com/apache/incubator-openwhisk/issues/2443

### CODE REVIEW

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 1]
```diff
@@ -230,6 +230,25 @@ trait WskWebActionsTests
...
+            val name = "webaction"
+            val file = Some(TestUtils.getTestActionFilename("corsHeaderMod.js"))
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) =>
+                    action.create(name, file, web = Some("true"))
+            }
+
+            val host = getServiceURL()
+            val url = host + s"$testRoutePath/$namespace/default/webaction.http"
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: want to test this for non `.http` extension also.

Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: that is for both .http and .json, the controller should be adding the expected headers for access control. 
furthermore, let's make sure we're not overriding and removing other headers and the test action should have other headers included.

File: tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala [hunk 2]
```diff
@@ -1133,11 +1133,16 @@ trait WebActionsApiTests extends ControllerTestCommon with BeforeAndAfterEach wi
 
             Seq(s"$systemId/proxy/export_c.http").
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala]: add to the seq non .http extensions.

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala (hunk 1 above)
Comment (dubeejw) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: Here I'm just doing an integration test for an options route without the web-custom-options annotation enabled. Added another test for `.json` below. Can combined the two tesets.

Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: ok dont need to combine i might have just missed the test. 
i'll take a closer look.

**Comment by dubeejw** [comment]
PG3 802.

File: tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala [hunk 3]
```diff
@@ -1131,13 +1131,18 @@ trait WebActionsApiTests extends ControllerTestCommon with BeforeAndAfterEach wi
             implicit val tid = transid()
             customOptions = false
 
-            Seq(s"$systemId/proxy/export_c.http").
+            Seq(s"$systemId/proxy/export_c.http", s"$systemId/proxy/export_c.json?a=b&c=d").
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala]: the query prams are not needed here - copy paste bug?

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 4]
```diff
@@ -215,19 +215,46 @@ trait WskWebActionsTests
...
+            val host = getServiceURL()
+            val url = host + s"$testRoutePath/$namespace/default/webaction"
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) =>
+                    action.create(name, file, web = Some("true"))
+            }
+
+            val responses = Seq(RestAssured.given().config(sslconfig).options(s"$url.http"),
+                RestAssured.given().config(sslconfig).get(s"$url.json"))
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: i find this a little easier to parse:

```
            val responses = Seq(
                RestAssured.given().config(sslconfig).options(s"$url.http"),
                RestAssured.given().config(sslconfig).get(s"$url.json"))
```

**Pull request merged**